v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/metadata/CassandraSourceMetadata.java

```diff
@@ -0,0 +1,245 @@
+
+  private CassandraSourceMetadata(ResultSet resultSet, Schema schema) {
+    this.resultSet = resultSet;
+    this.schema = schema;
+  }
+
+  /**
+   * Generates a {@link SourceSchema} from a Cassandra {@link ResultSet}.
+   *
+   * @return A {@link SourceSchema} instance representing the schema of the Cassandra source.
+   */
+  public SourceSchema generateSourceSchema() {
+    Map<String, Map<String, SourceColumn>> schemaMap = new HashMap<>();
+
+    resultSet.forEach(
+        row -> {
+          String tableName = row.getString("table_name");
+          String columnName = row.getString("column_name");
+          String dataType = row.getString("type");
+          String kind = row.getString("kind");
+
+          boolean isPrimaryKey = isPrimaryKey(kind);
+          SourceColumn sourceColumn = SourceColumn.create(columnName, kind, dataType, isPrimaryKey);
+
+          schemaMap.computeIfAbsent(tableName, k -> new HashMap<>()).put(columnName, sourceColumn);
+        });
+
+    Map<String, SourceTable> tables =
+        schemaMap.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey,
+                    entry ->
+                        SourceTable.create(
+                            entry.getKey(), ImmutableList.copyOf(entry.getValue().values()))));
+
+    return SourceSchema.create(Map.copyOf(tables));
+  }
+
+  /**
+   * Converts a {@link ResultSet} to a {@link Schema} object, updating the provided schema with the
+   * transformed Cassandra schema.
+   */
+  public void generateAndSetSourceSchema() {
+    SourceSchema sourceSchema = generateSourceSchema();
+    Map<String, com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable> sourceTableMap =
+        convertSourceSchemaToMap(sourceSchema);
+    schema.setSrcSchema(sourceTableMap);
+    schema.setToSource(convertSourceToNameAndColsTable(sourceSchema.tables().values()));
+  }
+
+  /**
+   * Converts a {@link SourceSchema} to a map of Spanner table names to {@link SourceTable} objects.
+   *
+   * @param sourceSchema The SourceSchema to convert.
+   * @return A map where the key is the table name and the value is the corresponding {@link
+   *     SourceTable}.
+   */
+  private Map<String, com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable>
+      convertSourceSchemaToMap(SourceSchema sourceSchema) {
+    return sourceSchema.tables().entrySet().stream()
+        .collect(
+            Collectors.toMap(
+                Map.Entry::getKey, entry -> convertSourceTableToSourceTable(entry.getValue())));
+  }
+
+  /**
+   * Converts a {@link SourceTable} to a {@link SourceTable} for Spanner.
+   *
+   * @param sourceTable The SourceTable to convert.
+   * @return A converted {@link SourceTable} object suitable for Spanner.
+   */
+  private com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable
+      convertSourceTableToSourceTable(SourceTable sourceTable) {
```

Comment:
Do you mean  `convertSourceTableToSpannerTable`? The method name and comment is very confusing

Reply:
Yes @shreyakhajanchi I have updated the Javadoc as well as Method Name